spec/ajv.spec.js

```diff
@@ -377,26 +377,70 @@ describe('Ajv', function () {
 
 
   describe('addFormat method', function() {
-    it('should add format as regular expression', function() {
-      ajv.addFormat('identifier', /^[a-z_$][a-z0-9_$]*$/i);
-      testFormat();
-    });
+    describe.skip('without any type restrictions specified', function() {
```

Comment:
By the way, rules are called based on the data type, not on the presence of type keyword. So the way you did it (format keyword added to two types) it there can be some code added for both string and integer data type and the code in format.jst should skip adding code if type is incorrect (I think you're doing it already). So I am not sure I understand why you think that type keyword should be explicitly specified for it to work.